### PR TITLE
LPS-27092

### PR DESCRIPTION
**Pull request by Soumia-tamrani**

### CODE REVIEW

**Comment by pedroamarall** [comment]
ci:test:sf

**Comment by pedroamarall** [comment]
> ### ❌ ci:test:sf - 0 out of 1 jobs passed in 11 minutes
> Click here for more details.
> #### Base Branch:
> Branch Name: [master](https://github.com/liferay/liferay-portal/tree/master)Branch GIT ID: [ecad23f941d2361f0026f2174d7d5c394f2398e9](https://github.com/liferay/liferay-portal/commit/ecad23f941d2361f0026f2174d7d5c394f2398e9)
> 
> #### Sender Branch:
> Branch Name: [LPS-27092](https://github.com/Soumia-tamrani/liferay-portal/tree/LPS-27092)Branch GIT ID: [daeb59da0f5feabd4ba38d4c5d28081f852d187a](https://github.com/Soumia-tamrani/liferay-portal/commit/daeb59da0f5feabd4ba38d4c5d28081f852d187a)
> 
> 0 out of 1jobs PASSED
> #### 1 Failed Jobs:
> * [test-portal-source-format](https://test-1-6.liferay.com/job/test-portal-source-format/6727/)
> 
> ##### For more details click [here](https://test-1-6.liferay.com/userContent/jobs/test-portal-source-format/builds/6727/jenkins-report.html).
> POSHI Validation Failure
> 
>      [exec]  errors in POSHI
>      [exec] 
>      [exec] 
>      [exec] Invalid path locator ControlMenuPreviewPanel#CLOSE_PANEL at:
>      [exec] /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/macros/Mytestt/Testy.macro:8
>      [exec]    1 |definition \{
>      [exec]    2 |
>      [exec]    3 |    macro Applycustomsize \{
>      [exec]    4 |        Click(locator1 = "ControlMenuPreviewPanel#APPLY_CUSTOM_SIZE");
>      [exec]    5 |    \}
>      [exec]    6 |
>      [exec]    7 |    macro ClosePanel \{
>      [exec]  > 8 |        Click(locator1 = "ControlMenuPreviewPanel#CLOSE_PANEL");
>      [exec]    9 |    \}
>      [exec]   10 |
>      [exec]   11 |    macro CreateNewForm \{
>      [exec]   12 |        FormsAdmin.addForm();
>      [exec]   13 |
>      [exec]   14 |        Form.editName(formName = "First Form");
>      [exec]   15 |
>      [exec] 
>      [exec] 
>      [exec] 
>      [exec] 
>      [exec] Invalid path locator ControlMenuPreviewPanel#VALIDATE_VALUES at:
>      [exec] /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/macros/Mytestt/Testy.macro:78
>      [exec]   71 |
>      [exec]   72 |        Form.editFieldLabelMetal(fieldValue = "??ge");
>      [exec]   73 |
>      [exec]   74 |        Form.save();
>      [exec]   75 |    \}
>      [exec]   76 |
>      [exec]   77 |    macro ValidateApply \{
>      [exec] > 78 |        AssertTextEquals(
>      [exec]   79 |            locator1 = "ControlMenuPreviewPanel#VALIDATE_VALUES",
>      [exec]   80 |            value1 = "Custom Size 1920x1080 Applied");
>      [exec]   81 |    \}
>      [exec]   82 |
>      [exec]   83 |    macro ValidateFormPublish \{
>      [exec]   84 |        Form.publishForm();
>      [exec]   85 |
>      [exec] 
>      [exec] 
>      [exec] 
>      [exec] 
>      [exec] Invalid path locator ControlMenuPreviewPanel#APPLY_CUSTOM_SIZE at:
>      [exec] /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/macros/Mytestt/Testy.macro:4
>      [exec]    1 |definition \{
>      [exec]    2 |
>      [exec]    3 |    macro Applycustomsize \{
>      [exec]  > 4 |        Click(locator1 = "ControlMenuPreviewPanel#APPLY_CUSTOM_SIZE");
>      [exec]    5 |    \}
>      [exec]    6 |
>      [exec]    7 |    macro ClosePanel \{
>      [exec]    8 |        Click(locator1 = "ControlMenuPreviewPanel#CLOSE_PANEL");
>      [exec]    9 |    \}
>      [exec]   10 |
>      [exec]   11 |    macro CreateNewForm \{
>      [exec] 
>      [exec] 
>      [exec] 
>      [exec] 
>      [exec] 
>      [exec] 
>      [exec] > Task :validatePoshi FAILED

Hey @Soumia-tamrani thanks for this pr, Can you fix your path file, The suite from this suite show that you don't have some Paths that you call in your test case

**Comment by Soumia-tamrani** [comment]
> > ### ❌ ci:test:sf - 0 out of 1 jobs passed in 11 minutes
> > Click here for more details.
> > #### Base Branch:
> > Branch Name: [master](https://github.com/liferay/liferay-portal/tree/master)Branch GIT ID: [ecad23f941d2361f0026f2174d7d5c394f2398e9](https://github.com/liferay/liferay-portal/commit/ecad23f941d2361f0026f2174d7d5c394f2398e9)
> > #### Sender Branch:
> > Branch Name: [LPS-27092](https://github.com/Soumia-tamrani/liferay-portal/tree/LPS-27092)Branch GIT ID: [daeb59da0f5feabd4ba38d4c5d28081f852d187a](https://github.com/Soumia-tamrani/liferay-portal/commit/daeb59da0f5feabd4ba38d4c5d28081f852d187a)
> > 0 out of 1jobs PASSED
> > #### 1 Failed Jobs:
> > 
> > * [test-portal-source-format](https://test-1-6.liferay.com/job/test-portal-source-format/6727/)
> > 
> > ##### For more details click [here](https://test-1-6.liferay.com/userContent/jobs/test-portal-source-format/builds/6727/jenkins-report.html).
> > POSHI Validation Failure
> > ```
> >  [exec]  errors in POSHI
> >  [exec] 
> >  [exec] 
> >  [exec] Invalid path locator ControlMenuPreviewPanel#CLOSE_PANEL at:
> >  [exec] /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/macros/Mytestt/Testy.macro:8
> >  [exec]    1 |definition \{
> >  [exec]    2 |
> >  [exec]    3 |    macro Applycustomsize \{
> >  [exec]    4 |        Click(locator1 = "ControlMenuPreviewPanel#APPLY_CUSTOM_SIZE");
> >  [exec]    5 |    \}
> >  [exec]    6 |
> >  [exec]    7 |    macro ClosePanel \{
> >  [exec]  > 8 |        Click(locator1 = "ControlMenuPreviewPanel#CLOSE_PANEL");
> >  [exec]    9 |    \}
> >  [exec]   10 |
> >  [exec]   11 |    macro CreateNewForm \{
> >  [exec]   12 |        FormsAdmin.addForm();
> >  [exec]   13 |
> >  [exec]   14 |        Form.editName(formName = "First Form");
> >  [exec]   15 |
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] Invalid path locator ControlMenuPreviewPanel#VALIDATE_VALUES at:
> >  [exec] /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/macros/Mytestt/Testy.macro:78
> >  [exec]   71 |
> >  [exec]   72 |        Form.editFieldLabelMetal(fieldValue = "??ge");
> >  [exec]   73 |
> >  [exec]   74 |        Form.save();
> >  [exec]   75 |    \}
> >  [exec]   76 |
> >  [exec]   77 |    macro ValidateApply \{
> >  [exec] > 78 |        AssertTextEquals(
> >  [exec]   79 |            locator1 = "ControlMenuPreviewPanel#VALIDATE_VALUES",
> >  [exec]   80 |            value1 = "Custom Size 1920x1080 Applied");
> >  [exec]   81 |    \}
> >  [exec]   82 |
> >  [exec]   83 |    macro ValidateFormPublish \{
> >  [exec]   84 |        Form.publishForm();
> >  [exec]   85 |
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] Invalid path locator ControlMenuPreviewPanel#APPLY_CUSTOM_SIZE at:
> >  [exec] /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/macros/Mytestt/Testy.macro:4
> >  [exec]    1 |definition \{
> >  [exec]    2 |
> >  [exec]    3 |    macro Applycustomsize \{
> >  [exec]  > 4 |        Click(locator1 = "ControlMenuPreviewPanel#APPLY_CUSTOM_SIZE");
> >  [exec]    5 |    \}
> >  [exec]    6 |
> >  [exec]    7 |    macro ClosePanel \{
> >  [exec]    8 |        Click(locator1 = "ControlMenuPreviewPanel#CLOSE_PANEL");
> >  [exec]    9 |    \}
> >  [exec]   10 |
> >  [exec]   11 |    macro CreateNewForm \{
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] > Task :validatePoshi FAILED
> > ```
> 
> Hey @Soumia-tamrani thanks for this pr, Can you fix your path file, The suite from this suite show that you don't have some Paths that you call in your test case

Hey @pedroamarall Ok I will fix it right now thank'you.

**Comment by Soumia-tamrani** [comment]
> > ### ❌ ci:test:sf - 0 out of 1 jobs passed in 11 minutes
> > Click here for more details.
> > #### Base Branch:
> > Branch Name: [master](https://github.com/liferay/liferay-portal/tree/master)Branch GIT ID: [ecad23f941d2361f0026f2174d7d5c394f2398e9](https://github.com/liferay/liferay-portal/commit/ecad23f941d2361f0026f2174d7d5c394f2398e9)
> > #### Sender Branch:
> > Branch Name: [LPS-27092](https://github.com/Soumia-tamrani/liferay-portal/tree/LPS-27092)Branch GIT ID: [daeb59da0f5feabd4ba38d4c5d28081f852d187a](https://github.com/Soumia-tamrani/liferay-portal/commit/daeb59da0f5feabd4ba38d4c5d28081f852d187a)
> > 0 out of 1jobs PASSED
> > #### 1 Failed Jobs:
> > 
> > * [test-portal-source-format](https://test-1-6.liferay.com/job/test-portal-source-format/6727/)
> > 
> > ##### For more details click [here](https://test-1-6.liferay.com/userContent/jobs/test-portal-source-format/builds/6727/jenkins-report.html).
> > POSHI Validation Failure
> > ```
> >  [exec]  errors in POSHI
> >  [exec] 
> >  [exec] 
> >  [exec] Invalid path locator ControlMenuPreviewPanel#CLOSE_PANEL at:
> >  [exec] /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/macros/Mytestt/Testy.macro:8
> >  [exec]    1 |definition \{
> >  [exec]    2 |
> >  [exec]    3 |    macro Applycustomsize \{
> >  [exec]    4 |        Click(locator1 = "ControlMenuPreviewPanel#APPLY_CUSTOM_SIZE");
> >  [exec]    5 |    \}
> >  [exec]    6 |
> >  [exec]    7 |    macro ClosePanel \{
> >  [exec]  > 8 |        Click(locator1 = "ControlMenuPreviewPanel#CLOSE_PANEL");
> >  [exec]    9 |    \}
> >  [exec]   10 |
> >  [exec]   11 |    macro CreateNewForm \{
> >  [exec]   12 |        FormsAdmin.addForm();
> >  [exec]   13 |
> >  [exec]   14 |        Form.editName(formName = "First Form");
> >  [exec]   15 |
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] Invalid path locator ControlMenuPreviewPanel#VALIDATE_VALUES at:
> >  [exec] /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/macros/Mytestt/Testy.macro:78
> >  [exec]   71 |
> >  [exec]   72 |        Form.editFieldLabelMetal(fieldValue = "??ge");
> >  [exec]   73 |
> >  [exec]   74 |        Form.save();
> >  [exec]   75 |    \}
> >  [exec]   76 |
> >  [exec]   77 |    macro ValidateApply \{
> >  [exec] > 78 |        AssertTextEquals(
> >  [exec]   79 |            locator1 = "ControlMenuPreviewPanel#VALIDATE_VALUES",
> >  [exec]   80 |            value1 = "Custom Size 1920x1080 Applied");
> >  [exec]   81 |    \}
> >  [exec]   82 |
> >  [exec]   83 |    macro ValidateFormPublish \{
> >  [exec]   84 |        Form.publishForm();
> >  [exec]   85 |
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] Invalid path locator ControlMenuPreviewPanel#APPLY_CUSTOM_SIZE at:
> >  [exec] /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/macros/Mytestt/Testy.macro:4
> >  [exec]    1 |definition \{
> >  [exec]    2 |
> >  [exec]    3 |    macro Applycustomsize \{
> >  [exec]  > 4 |        Click(locator1 = "ControlMenuPreviewPanel#APPLY_CUSTOM_SIZE");
> >  [exec]    5 |    \}
> >  [exec]    6 |
> >  [exec]    7 |    macro ClosePanel \{
> >  [exec]    8 |        Click(locator1 = "ControlMenuPreviewPanel#CLOSE_PANEL");
> >  [exec]    9 |    \}
> >  [exec]   10 |
> >  [exec]   11 |    macro CreateNewForm \{
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] 
> >  [exec] > Task :validatePoshi FAILED
> > ```
> 
> Hey @Soumia-tamrani thanks for this pr, Can you fix your path file, The suite from this suite show that you don't have some Paths that you call in your test case

Hey @pedroamarall Ok, i will fix it right now thank'you :)

**Comment by Soumia-tamrani** [comment]
> > > ### ❌ ci:test:sf - 0 out of 1 jobs passed in 11 minutes
> > > Click here for more details.
> > > #### Base Branch:
> > > Branch Name: [master](https://github.com/liferay/liferay-portal/tree/master)Branch GIT ID: [ecad23f941d2361f0026f2174d7d5c394f2398e9](https://github.com/liferay/liferay-portal/commit/ecad23f941d2361f0026f2174d7d5c394f2398e9)
> > > #### Sender Branch:
> > > Branch Name: [LPS-27092](https://github.com/Soumia-tamrani/liferay-portal/tree/LPS-27092)Branch GIT ID: [daeb59da0f5feabd4ba38d4c5d28081f852d187a](https://github.com/Soumia-tamrani/liferay-portal/commit/daeb59da0f5feabd4ba38d4c5d28081f852d187a)
> > > 0 out of 1jobs PASSED
> > > #### 1 Failed Jobs:
> > > 
> > > * [test-portal-source-format](https://test-1-6.liferay.com/job/test-portal-source-format/6727/)
> > > 
> > > ##### For more details click [here](https://test-1-6.liferay.com/userContent/jobs/test-portal-source-format/builds/6727/jenkins-report.html).
> > > POSHI Validation Failure
> > > ```
> > >  [exec]  errors in POSHI
> > >  [exec] 
> > >  [exec] 
> > >  [exec] Invalid path locator ControlMenuPreviewPanel#CLOSE_PANEL at:
> > >  [exec] /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/macros/Mytestt/Testy.macro:8
> > >  [exec]    1 |definition \{
> > >  [exec]    2 |
> > >  [exec]    3 |    macro Applycustomsize \{
> > >  [exec]    4 |        Click(locator1 = "ControlMenuPreviewPanel#APPLY_CUSTOM_SIZE");
> > >  [exec]    5 |    \}
> > >  [exec]    6 |
> > >  [exec]    7 |    macro ClosePanel \{
> > >  [exec]  > 8 |        Click(locator1 = "ControlMenuPreviewPanel#CLOSE_PANEL");
> > >  [exec]    9 |    \}
> > >  [exec]   10 |
> > >  [exec]   11 |    macro CreateNewForm \{
> > >  [exec]   12 |        FormsAdmin.addForm();
> > >  [exec]   13 |
> > >  [exec]   14 |        Form.editName(formName = "First Form");
> > >  [exec]   15 |
> > >  [exec] 
> > >  [exec] 
> > >  [exec] 
> > >  [exec] 
> > >  [exec] Invalid path locator ControlMenuPreviewPanel#VALIDATE_VALUES at:
> > >  [exec] /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/macros/Mytestt/Testy.macro:78
> > >  [exec]   71 |
> > >  [exec]   72 |        Form.editFieldLabelMetal(fieldValue = "??ge");
> > >  [exec]   73 |
> > >  [exec]   74 |        Form.save();
> > >  [exec]   75 |    \}
> > >  [exec]   76 |
> > >  [exec]   77 |    macro ValidateApply \{
> > >  [exec] > 78 |        AssertTextEquals(
> > >  [exec]   79 |            locator1 = "ControlMenuPreviewPanel#VALIDATE_VALUES",
> > >  [exec]   80 |            value1 = "Custom Size 1920x1080 Applied");
> > >  [exec]   81 |    \}
> > >  [exec]   82 |
> > >  [exec]   83 |    macro ValidateFormPublish \{
> > >  [exec]   84 |        Form.publishForm();
> > >  [exec]   85 |
> > >  [exec] 
> > >  [exec] 
> > >  [exec] 
> > >  [exec] 
> > >  [exec] Invalid path locator ControlMenuPreviewPanel#APPLY_CUSTOM_SIZE at:
> > >  [exec] /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/macros/Mytestt/Testy.macro:4
> > >  [exec]    1 |definition \{
> > >  [exec]    2 |
> > >  [exec]    3 |    macro Applycustomsize \{
> > >  [exec]  > 4 |        Click(locator1 = "ControlMenuPreviewPanel#APPLY_CUSTOM_SIZE");
> > >  [exec]    5 |    \}
> > >  [exec]    6 |
> > >  [exec]    7 |    macro ClosePanel \{
> > >  [exec]    8 |        Click(locator1 = "ControlMenuPreviewPanel#CLOSE_PANEL");
> > >  [exec]    9 |    \}
> > >  [exec]   10 |
> > >  [exec]   11 |    macro CreateNewForm \{
> > >  [exec] 
> > >  [exec] 
> > >  [exec] 
> > >  [exec] 
> > >  [exec] 
> > >  [exec] 
> > >  [exec] > Task :validatePoshi FAILED
> > > ```
> > 
> > 
> > Hey @Soumia-tamrani thanks for this pr, Can you fix your path file, The suite from this suite show that you don't have some Paths that you call in your test case
> 
> Hey @pedroamarall Ok, i will fix it right now thank'you :)

Hey @pedroamarall I fixed my path file, Can you check my new PR Updated !!

**Comment by pedroamarall** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/MyBlogTests/MyBlogTests.testcase [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+				portlet = "Blogs");
+		}
+
+		task ("And the first blog entry added") {
+			Blogs.addEntry(
+				entryContent = "1st Blog",
+				entryTitle = "Blog1");
+		}
+
+		task ("Blog add sucessfuly") {
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/tests/MyBlogTests/MyBlogTests.testcase:133]: This task is not necessary, you can put inside the above task

File: portal-web/test/functional/com/liferay/portalweb/tests/MyBlogTests/MyBlogTests.testcase [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	test CanSearchBlogs {
+		task ("Given the Blogs Portlet is added to the page") {
+			Navigator.openURL();
+
+			ProductMenu.gotoPortlet(
+				category = "Content & Data",
+				portlet = "Blogs");
+		}
+
+		task ("And the first blog entry added") {
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/tests/MyBlogTests/MyBlogTests.testcase:127]: you can edit to ( And Given the 2 blogs entry is added)

File: portal-web/test/functional/com/liferay/portalweb/tests/MyBlogTests/MyBlogTests.testcase [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+			Blogs.viewEntryPresent(entryTitle = "Blog1");
+		}
+
+		task ("And the second blog entry added") {
+			Blogs.addEntry(
+				entryContent = "soumia",
+				entryTitle = "soumiatamrani");
+		}
+
+		task ("Blog add sucessfuly") {
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/tests/MyBlogTests/MyBlogTests.testcase:143]: This task is not necessary, you can put inside the above task

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyTestFile.testcase [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+			Click(
+				key_deviceType = "Custom",
+				locator1 = "ControlMenuPreviewPanel#PREVIEW_DATA_DEVICE_BUTTON");
+		}
+
+		task ("Then assert that the values defined in step 4 persisted") {
+			Testy.ValidateApply();
+		}
+	}
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyTestFile.testcase:52]: I think the test is not exactly that, could you check the test function again please

**Pull request closed without merging**